### PR TITLE
Docs quick tutorial update

### PR DESCRIPTION
**Pull request by stevepiercy**
This PR (hopefully) eliminates the common gotcha where users jump into later steps without installing requirements in earlier steps. 

- `setup.py` includes all requirements as needed.
- Narrative text no longer mentions manual installation of dependencies.
- Standardize formatting with Documentation Style Guide.
- Setuptools updates for its glossary term.

### CODE REVIEW

**Comment by mmerickel** [comment]
I would have really preferred if you had used `testing` as the name of the extra to be consistent with every repository we maintain as well as the pattern set forth in the cookiecutters.

**Comment by stevepiercy** [comment]
No problem. I'll make changes in another PR, along with moving the pyramid_debugtoolbar into `dev` in extras.

File: docs/quick_tutorial/functional_testing.rst [hunk 1]
```diff
@@ -31,31 +31,40 @@ Objectives
...
+        :emphasize-lines: 14
 
-   Be sure this file is not executable, or ``pytest`` may not include your
-   tests.
+#.  Install our project and its newly added dependency.
+    Note that we use the extra specifier ``[test]`` to install testing requirements.
+
+    .. code-block:: bash
+
+        $VENV/bin/pip install -e .[test]
```
Comment (mmerickel) [inline docs/quick_tutorial/functional_testing.rst]: This should use quotes, such as `pip install -e ".[test]"`.

File: docs/quick_tutorial/jinja2/setup.py [hunk 2]
```diff
@@ -2,13 +2,22 @@
...
+              'webtest',
+          ],
+      },
       entry_points="""\
       [paste.app_factory]
       main = tutorial:main
       """,
-)
\ No newline at end of file
+      )
```
Comment (mmerickel) [inline docs/quick_tutorial/jinja2/setup.py]: Why are all these closing parenthesis indented? They should not be.

Comment (stevepiercy) [reply docs/quick_tutorial/jinja2/setup.py]: PyCharm has an inspection for PEP8. Either indentation form is acceptable.
https://www.python.org/dev/peps/pep-0008/#indentation

**Comment by stevepiercy** [comment]
Revisions are now in PR #3380.

**Pull request merged**